gunicorn/workers/gthread.py

```diff
@@ -205,33 +207,37 @@ def run(self):
             # can we accept more connections?
             if self.nr_conns < self.worker_connections:
                 # wait for an event
-                events = self.poller.select(0.02)
+                events = self.poller.select(1.0)
```

Comment:
It will return as soon as the first socket is ready. There is no danger
such as you describe.

On Fri, Mar 25, 2016, 03:10 Benoit Chesneau <notifications@github.com>
wrote:

> In gunicorn/workers/gthread.py
> <https://github.com/benoitc/gunicorn/pull/1230#discussion_r57432600>:
>
> > @@ -205,33 +207,37 @@ def run(self):
> >              # can we accept more connections?
> >              if self.nr_conns < self.worker_connections:
> >                  # wait for an event
> > -                events = self.poller.select(0.02)
> > +                events = self.poller.select(1.0)
>
> the reason to be faster there is that this selector is done on read to
> check if the socket enter in a state where we can accept. If we wait too
> much we can queue more accepts than we want and we fail to balance
> correctly the connections between threads and workers. So we should be fast
> there imo.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/benoitc/gunicorn/pull/1230/files/f2418a95e0df21ae53fc8dba96995e18b2efa961#r57432600>
>